assets/js/src/newsletter_editor/components/history.js

```diff
@@ -44,7 +44,7 @@ Module.HistoryView = Marionette.View.extend({
     if (this.model.currentStateIndex > 0) {
       this.model.states.splice(0, this.model.currentStateIndex);
     }
-    this.model.states.unshift(json.body);
+    this.model.states.unshift(JSON.stringify(json.body));
```

Comment:
Does this really need to be stringified (and later parsed I suppose)?

Reply:
In later commits, I added a comparison of the current state with the one being saved (so you won't create same states just by clicking on Save button), and for that I need both states stringified.